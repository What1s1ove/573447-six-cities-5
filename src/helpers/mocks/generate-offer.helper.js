import {getRandomItem} from '~/helpers/array/array';
import {getRandomNumber} from '~/helpers/number/number';
import {generateRandomId} from '~/helpers/mocks/generate-random-id.helper';
import {OfferCity, OfferType} from '~/common/enums/enums';

const TITLES = [
  `Beautiful \u0026 luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];

const OfferConfig = {
  PRICE: {
    MIN: 50,
    MAX: 250,
  },
  RATING: {
    MIN: 0,
    MAX: 5,
    FRACTIONAL_PART_COUNT: 1,
  },
};

const cities = Object.values(OfferCity);
const offerTypes = Object.values(OfferType);

const generateOffer = () => ({
  id: generateRandomId(),
  city: getRandomItem(cities),
  type: getRandomItem(offerTypes),
  title: getRandomItem(TITLES),
  price: getRandomNumber(OfferConfig.PRICE.MIN, OfferConfig.PRICE.MAX),
  rating: getRandomNumber(
      OfferConfig.RATING.MIN,
      OfferConfig.RATING.MAX,
      OfferConfig.RATING.FRACTIONAL_PART_COUNT
  ),
});

export {generateOffer};
