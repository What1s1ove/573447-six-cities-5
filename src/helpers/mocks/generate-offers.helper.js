import {generateOffer} from '~/helpers/mocks/generate-offer.helper';

const generateOffers = (count) => {
  const offers = Array.from(new Array(count), generateOffer);

  return offers;
};

export {generateOffers};
