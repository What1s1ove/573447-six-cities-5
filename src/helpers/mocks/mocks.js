export * from './generate-offer.helper';
export * from './generate-offers.helper';
export * from './generate-random-id.helper';
