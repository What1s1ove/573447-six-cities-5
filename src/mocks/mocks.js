export * from './offers/offers';
export * from './user/user';
export * from './reviews/reviews';
export * from './city/city';
export * from './store/store';
