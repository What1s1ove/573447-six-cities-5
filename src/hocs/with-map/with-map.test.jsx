import * as React from 'react';
import {render} from 'enzyme';
import {mockedCity, mockedOffer, mockedOffers} from '~/mocks/mocks';
import withMap from './with-map';

const MockedComponent = () => <div />;
const WrappedMockedComponent = withMap(MockedComponent);

describe(`withMap hoc`, () => {
  it(`should correctly output data from withMap hoc`, () => {
    const wrapper = render(
        <WrappedMockedComponent
          city={mockedCity}
          activeOffer={mockedOffer}
          offers={mockedOffers}
        />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
