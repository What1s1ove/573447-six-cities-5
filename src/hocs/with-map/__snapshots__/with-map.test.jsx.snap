// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMap hoc should correctly output data from withMap hoc 1`] = `<div />`;
